lib/Parse/Lexer.cpp

```diff
@@ -1843,9 +1843,21 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
   assert((QuoteChar == '"' || QuoteChar == '\'') && "Unexpected start");
 
   bool IsMultilineString = advanceIfMultilineDelimiter(CurPtr, Diags);
-  if (IsMultilineString && *CurPtr != '\n' && *CurPtr != '\r')
-    diagnose(CurPtr, diag::lex_illegal_multiline_string_start)
+  // Test for single-line Strings that may resemble multiline delimiter
+  if (IsMultilineString && *CurPtr != '\n' && *CurPtr != '\r') {
+    const char *TmpPtr = CurPtr + 1;
+    TmpPtr = skipToEndOfInterpolatedExpression(TmpPtr, BufferEnd, false);
```

Comment:
I don't think `skipToEndOfInterpolatedExpression()` work for this. That's for, for example, given `"foo \(x + y + z) bar"`, skipping from `x` to right after `z`.
Instead, you should manually search `"` by advancing `TmpPtr`.

Reply:
Changed the logic to iterate through the buffer. Thank you for the guidance!